engine/src/witness/sol.rs

```diff
@@ -140,6 +141,26 @@ impl VoterApi<SolanaEgressWitnessing> for SolanaEgressWitnessingVoter {
 	}
 }
 
+#[allow(dead_code)]
+#[derive(Clone)]
+struct SolanaVaultSwapsVoter {
+	client: SolRetryRpcClient,
+}
+
+#[async_trait::async_trait]
+impl VoterApi<SolanaVaultSwapTracking> for SolanaVaultSwapsVoter {
+	async fn vote(
+		&self,
+		_settings: <SolanaVaultSwapTracking as ElectoralSystem>::ElectoralSettings,
+		_properties: <SolanaVaultSwapTracking as ElectoralSystem>::ElectionProperties,
+	) -> Result<
+		<<SolanaVaultSwapTracking as ElectoralSystem>::Vote as VoteStorage>::Vote,
+		anyhow::Error,
+	> {
+		todo!()
```

Comment:
This should be implemented alongside the ElectoralSystem, is there a reason you haven't done so here? It's kind of impossible to tell if the electoral system is appropriate if this side of it isn't implemented

Reply:
Albert is doing the engine side so I left it as a todo. Gonna merge albert's PR into this and hook up the elections to it as discussed in discord.